tools/hardware/utils/memory.py

```diff
@@ -26,6 +26,38 @@ def visitor(node: WrappedNode):
     return regions
 
 
+def merge_memory_regions(regions: Set[Region]) -> Set[Region]:
+    ''' Check all region and merge adjacent ones '''
+    all_regions = []
+    idx = 0
+    for region in regions:
```

Comment:
prefer enumerate here

Reply:
What do you mean? Get everything done in this loop?

Reply:
```suggestion
    for idx, region in enumerate(regions):
```

Reply:
Then you don't need to worry about incrementing or declaring the idx variable

Reply:
Yeap, sorry I'm a bit old school on this :)